drivers/ieee802154/ieee802154_rf2xx.c

```diff
@@ -66,47 +67,15 @@ static struct net_pkt rf2xx_ack_pkt = {
 };
 #endif /* CONFIG_NET_L2_OPENTHREAD */
 
-/**
- * RF output power for RF2xx
- *
- * The table below is exact for RF233. For RF231/2 the TX power might
- * be a bit off, but good enough.
- *
- * RF233: http://ww1.microchip.com/downloads/en/devicedoc/atmel-8351-mcu_wireless-at86rf233_datasheet.pdf
- * 9.2.5 Register Description Register 0x05 (PHY_TX_PWR)
- * 0x0 = 4dBm .. 0xF = -17dBm
- *
- * RF232: http://ww1.microchip.com/downloads/en/DeviceDoc/doc8321.pdf
- * 9.2.5 Register Description Register 0x05 (PHY_TX_PWR)
- * 0x0 = 3dBm .. 0xF = -17dBm
- *
- * RF231: http://ww1.microchip.com/downloads/en/DeviceDoc/doc8111.pdf
- * 9.2.5 Register Description Register 0x05 (PHY_TX_PWR)
- * 0x0 = 3dBm .. 0xF = -17dBm
- */
+#define TX_PWR_TABLE_SIZE DT_INST_PROP_LEN(0, tx_pwr_table)
 
-#define RF2XX_OUTPUT_POWER_MAX		4
-#define RF2XX_OUTPUT_POWER_MIN		(-17)
-
-/* Lookup table for PHY_TX_PWR register for RF233 */
-static const uint8_t phy_tx_pwr_lt[] = {
-	0xf,                     /* -17  dBm: -17 */
-	0xe, 0xe, 0xe, 0xe, 0xe, /* -12  dBm: -16, -15, -14, -13, -12 */
-	0xd, 0xd, 0xd, 0xd,      /* -8   dBm: -11, -10, -9, -8 */
-	0xc, 0xc,                /* -6   dBm: -7, -6 */
-	0xb, 0xb,                /* -4   dBm: -5, -4 */
-	0xa,                     /* -3   dBm: -3 */
-	0x9,                     /* -2   dBm: -2 */
-	0x8,                     /* -1   dBm: -1 */
-	0x7,                     /*  0.0 dBm:  0 */
-	0x6,                     /*  1   dBm:  1 */
-	0x5,                     /*  2   dBm:  2 */
-	/* 0x4, */               /*  2.5 dBm */
-	0x3,                     /*  3   dBm:  3 */
-	/* 0x2, */               /*  3.4 dBm */
-	/* 0x1, */               /*  3.7 dBm */
-	0x0                      /*  4   dBm: 4 */
-};
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) > 1
+#define THE_TABLE_SIZE(inst) DT_INST_PROP_LEN(inst, tx_pwr_table)
+#define CHECK_TABLE_SIZE(inst) \
+	BUILD_ASSERT(THE_TABLE_SIZE(inst) == TX_PWR_TABLE_SIZE, \
+		     "The TX PWR TABLE must be same size for all instances.");
+DT_INST_FOREACH_STATUS_OKAY(CHECK_TABLE_SIZE)
```

Comment:
note: you can allow multiple sizes by just defining the table outside of driver's config/data, and just take a pointer to the table + its size instead. See for example https://github.com/teslabs/spinner/blob/5591731a547a9d5ee50636360122e53625cacb4b/drivers/svpwm/svpwm_stm32.c#L298-L299